src/core/abr/representation_chooser.ts

```diff
@@ -449,8 +501,39 @@ export default class RepresentationChooser {
             nextBitrate /= clock.speed;
           }
 
+          if ((this.currentDroppedFrames / this.currentTotalFrames) > 0.33) {
+            if (clock.bitrate) {
+              const decodableBitrate =
+                getDecodableBitrate(representations, clock.bitrate);
+              if (decodableBitrate < this._decodableBitrate) {
+                this._decodableBitrate = decodableBitrate;
+                if (this._timeOutId !== null) {
+                  clearTimeout(this._timeOutId);
+                }
+                this._timeOutId = window.setTimeout(() => {
+                  this._decodableBitrate = Infinity;
+                  this._timeOutId = null;
+                  this._timeOutDuration =
+                    Math.min(Math.pow(this._timeOutDuration, 2), 32);
+                }, this._timeOutDuration * 60 * 1000);
+              }
+            }
+          }
+
+          const filters = deviceEvents;
+
+          if (deviceEvents.hasOwnProperty("bitrate")) {
```

Comment:
Mutating it likes that makes me uncomfortable...

Maybe we should not work around it like that here.